packages/flutter_tools/lib/src/compile.dart

```diff
@@ -110,11 +118,19 @@ class StdoutHandler {
         return;
       }
       final int spaceDelimiter = message.lastIndexOf(' ');
-      compilerOutput.complete(
-          CompilerOutput(
-              message.substring(boundaryKey.length + 1, spaceDelimiter),
-              int.parse(message.substring(spaceDelimiter + 1).trim()),
-              sources));
+      final String fileName = message.substring(boundaryKey.length + 1, spaceDelimiter);
+      final int errorCount = int.parse(message.substring(spaceDelimiter + 1).trim());
```

Comment:
Is there something reasonable to do to handle this string parse failing on one of these steps?

Reply:
We probably want this to be a fatal unhandled error so that we see it in crash logging if it ever occurs